test/tests/functional/pbs_reservations.py

```diff
@@ -1327,53 +1328,47 @@ def test_standing_resv_with_job_array(self):
         exp_attr = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
         self.server.expect(RESV, exp_attr, id=rid)
         rid_q = rid.split(".")[0]
-        # Submit a job-array within resrvation
-        j = Job(TEST_USER, attrs={ATTR_q: rid_q, ATTR_J: '1-5'})
-        j.set_sleep_time(40)
+        # Submit a job-array within reservation
+        j = Job(TEST_USER, attrs={'Resource_List.select': '1:ncpus=1',
+                                  ATTR_q: rid_q, ATTR_J: '1-4'})
+        j.set_sleep_time(15)
         jid = self.server.submit(j)
         subjid = []
-        for i in range(1, 5):
+        for i in range(1, 4):
             subjid.append(j.create_subjob_id(jid, i))
-        self.server.expect(JOB, {'job_state': 'Q',
-                                 'comment': (MATCH_RE,
-                                             'Queue not started')}, jid)
         # Wait for standing reservation first instance to start
         exp_attr = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
-        self.server.expect(RESV, exp_attr, id=rid, offset=10)
+        self.server.expect(RESV, exp_attr, id=rid)
         self.server.expect(RESV, {'reserve_index': 1}, id=rid)
         self.server.expect(JOB, {'job_state': 'B'}, jid)
         self.server.expect(JOB, {'job_state=R': 4}, count=True,
                            id=jid, extend='t')
-        self.server.expect(JOB, {'job_state=Q': 1}, count=True,
-                           id=jid, extend='t')
         # Wait for standing reservation first instance to finished
         self.logger.info(
-            'Waiting 60 seconds for first instance of reservation to Finish')
-        time.sleep(60)
+            'Waiting 20 seconds for first instance of reservation to Finish')
```

Comment:
why are you waiting for 20 seconds here? The last reservation was only 10 seconds long. So Ideally after 10 seconds it should get back to RESV_CONFIRMED state again.